### PR TITLE
OCPBUGS-4658: Apply shared defaulters to CRD-based routes.

### PR DESCRIPTION
**Pull request by benluddy**
This addresses the known remaining API disparities between CRD-based and openshift-apiserver-based Routes.

Proof PR: https://github.com/openshift/microshift/pull/1192.

### CODE REVIEW

File: openshift-kube-apiserver/admission/customresourcevalidation/route/default_route.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		// object.
+		return fmt.Errorf("object being admitted is of type %T and does not implement runtime.Unstructured", attributes.GetObject())
+	}
+
+	var external v1.Route
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.UnstructuredContent(), &external); err != nil {
+		return err
+	}
+
+	defaultScheme.Default(&external)
```
Comment (deads2k) [inline openshift-kube-apiserver/admission/customresourcevalidation/route/default_route.go:63]: why wire to scheme, why not directly call the defaulting logic you need?

Comment (benluddy) [reply openshift-kube-apiserver/admission/customresourcevalidation/route/default_route.go]: Good point, thanks. Done.

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**